### PR TITLE
feat(linter): Implement eslint(no-compare-neg-zero)

### PR DESCRIPTION
**Pull request by YangchenYe323**

### CODE REVIEW

File: crates/oxc_linter/src/rules/no_compare_neg_zero.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    /// ```
+    NoCompareNegZero,
+    correctness
+);
+
+impl Rule for NoCompareNegZero {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::BinaryExpression(expr) = node.get().kind() else { return; };
+        if let Some(op) =
+            OPERATORS_TO_CHECK.iter().find(|op| **op == expr.operator.as_str()).copied()
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/no_compare_neg_zero.rs:42]: Don't copy eslint on this one, we have all the tools we need ;-)

https://github.com/Boshen/oxc/blob/867f879483dfb7e3bcbda800db61cae55dcbbf35/crates/oxc_ast/src/ast/operator.rs#L184-L197

File: crates/oxc_linter/src/rules/no_compare_neg_zero.rs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    miette::{self, Diagnostic},
+    thiserror::{self, Error},
+};
+use oxc_macros::declare_oxc_lint;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-compare-neg-zero): Do not use the {0} operator to compare against -0.")]
+#[diagnostic(severity(warning))]
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/no_compare_neg_zero.rs:15]: I think we should have some kind of help here, why don't we want to compare against -0?

**Review by Boshen — CHANGES_REQUESTED** [review]

File: crates/oxc_linter/src/rules/no_compare_neg_zero.rs (hunk 2 above)
Comment (YangchenYe323) [reply crates/oxc_linter/src/rules/no_compare_neg_zero.rs]: Added help message. Does it look good to you?

File: crates/oxc_linter/src/rules/no_compare_neg_zero.rs (hunk 1 above)
Comment (YangchenYe323) [reply crates/oxc_linter/src/rules/no_compare_neg_zero.rs]: Done.

**Pull request merged**